src/iframe-helper.js

```diff
@@ -27,27 +27,36 @@ import {parseUrl} from './url';
  * @param {function(!Object)} callback Called when a message of this type
  *     arrives for this iframe.
  * @param {boolean=} opt_is3P set to true if the iframe is 3p.
+ * @param {string=} opt_messageTypePropName Property to use for type
+ *     checking (default: 'type').
+ * @param {boolean=} opt_noSentinel Skip sentinel check.
  * @return {!Unlisten}
  */
-export function listen(iframe, typeOfMessage, callback, opt_is3P) {
+export function listen(iframe, typeOfMessage, callback, opt_is3P,
+                       opt_messageTypePropName, opt_noSentinel) {
```

Comment:
You mean just don't use this listen method and use do a `win.addEventListener('message')` inside `amp-youtube`?